### PR TITLE
[@types/auth0] Add the missing audience property to the PasswordGrantOption. Add…

### PR DESCRIPTION
**Pull request by RobRolls**
… the missing refreshToken() for the AuthenticationClient and the OAuthAuthenticator.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
audience property:
https://auth0.com/docs/flows/call-your-api-using-resource-owner-password-flow#request-tokens
https://github.com/auth0/node-auth0/blob/master/src/auth/OAuthAuthenticator.js#L152

refreshToken:
https://github.com/auth0/node-auth0/blob/master/src/auth/OAuthAuthenticator.js#L210
https://github.com/auth0/node-auth0/blob/0441d66c0745fe482416d74232966efd6d3d83e0/src/auth/index.js#L532

### CODE REVIEW

**Comment by RobRolls** [comment]
Can anyone review this please?

**Comment by RobRolls** [comment]
Sorry to pester everyone but could I please get this reviewed? @westy92 @ianhowe76 @dauledk @dancrumb @pwrnrd @CatGuardian @MengRS @Scoup @isabela-morais

**Comment by RobRolls** [comment]
Bump?

**Comment by RobRolls** [comment]
Hi @westy92 @ianhowe76 @dauledk @dancrumb @pwrnrd @CatGuardian @MengRS @Scoup @isabela-morais

I wanted to wish you all a Happy Thanksgiving (if Thanksgiving is a holiday you celebrate). Thought I'd let you know that I will personally save you a slice of pie if you can find the time to review this PR. Getting the slice to you may prove a challenge, but I will certainly save it. :)

**Comment by andrewreineke** [comment]
+1

**Review by ianhowe76 — APPROVED** [review]

**Comment by RobRolls** [comment]
@ianhowe76 thanks for the review.  Would you mind re-reviewing this after my merge conflict fix.  Thanks!

File: types/auth0/index.d.ts [hunk 1]
```diff
@@ -611,13 +611,23 @@ export interface PasswordGrantOptions {
   password: string;
   realm?: string;
   scope?: string;
+  audience?: string;
```
Comment (peterblazejewicz) [inline types/auth0/index.d.ts:614]: cannot see `audience` as part of password grant options on the linked page?

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by RobRolls** [comment]
Ready to merge

**Pull request merged**